### PR TITLE
[SPARK-40654][PROTOBUF][FOLLOW-UP] Clean up SBT build in Proto component

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR cleans up the syntax, and properly copy protobuf assembly jar (currently it copies connect assembly jar by mistake).

### Why are the changes needed?

For consistent code style, and correct SBT build.

### Does this PR introduce _any_ user-facing change?
No, this isn't released yet.

### How was this patch tested?

CI in this PR should test it out.

### CODE REVIEW

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by LuciferYang** [comment]
hmm... should we add `assemblyPackageScala` and `assemblyExcludedJars` to `SparkProtobuf` module, the assembly jar is a little fat

```
[debug] Including from cache: spark-tags_2.12-3.4.0-SNAPSHOT.jar
[debug] Including from cache: unused-1.0.0.jar
[debug] Including from cache: spark-protobuf_2.12-3.4.0-SNAPSHOT.jar
[debug] Including from cache: scala-collection-compat_2.12-2.2.0.jar
[debug] Including from cache: pmml-model-1.4.8.jar
[debug] Including from cache: protobuf-java-3.21.1.jar
[debug] Including from cache: guava-14.0.1.jar
[debug] Including from cache: scala-library-2.12.17.jar
```

for example, similar as `SparkConnect` module:

```
// Exclude `scala-library` from assembly.
    (assembly / assemblyPackageScala / assembleArtifact) := false,

    // Exclude `pmml-model-*.jar`, `scala-collection-compat_*.jar`,
    // `spark-tags_*.jar`, "guava-*.jar" and `unused-1.0.0.jar` from assembly.
    (assembly / assemblyExcludedJars) := {
      val cp = (assembly / fullClasspath).value
      cp filter { v =>
        val name = v.data.getName
        name.startsWith("pmml-model-") || name.startsWith("scala-collection-compat_") ||
          name.startsWith("spark-tags_") || name.startsWith("guava-") || name == "unused-1.0.0.jar"
      }
    },
```

then the assembly jar just includes:

```
[debug] Including: spark-protobuf_2.12-3.4.0-SNAPSHOT.jar
[debug] Including from cache: protobuf-java-3.21.1.jar
```

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -703,32 +703,24 @@ object SparkConnect {
...
-    ),
+    dependencyOverrides += "com.google.protobuf" % "protobuf-java" % protoVersion,
 
     (Compile / PB.targets) := Seq(
-      PB.gens.java                -> (Compile / sourceManaged).value,
+      PB.gens.java -> (Compile / sourceManaged).value,
     ),
 
-    (assembly / test) := false,
+    (assembly / test) := { },
```
Comment (amaliujia) [inline project/SparkBuild.scala:723]: just for my own education, is this the same: `false` -> `{ }`? 

looks like a bit magic

Comment (HyukjinKwon) [reply project/SparkBuild.scala:723]: Yeah, they are same. I believe `{ }` is much more common (googled a bit)

**Review by amaliujia — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Let me get this in first in any event since it works (and the built is technically broken without this change).

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**